### DISCUSSION TITLE
Questions about the behavioral change of Persist middleware

### DISCUSSION THREAD

**Discussion by limuen** [pmndrs/zustand#2814] [Q&A]
Thank you very much to the author for creating such an excellent library as zustand. After upgrading to V5 version according to the documentation, even if I don't need it
useUserStore. setState (){
    token: null,
    loginToRoute: "setState"
});，
 But I can still get the default parameters I set locally. Is it because I'm using the wrong posture?
 
 repoUrl: https://github.com/limuen/monorepo-project/commit/5760d8af9f168f176080011eb0efb0751d5d0343#diff-224dbaccff558cc5afb7c40e26b5e9d729073d197b70dcb2788918d388532233
 
 Looking forward to your reply, thank you

**Comment by dbritto-dev** [comment]
@limuen I don't get your point

**Reply by limuen** [reply]
> @limuen我不明白你的意思

Doesn't the V5 version say that persists must have a setState? I can persist without setState

**Reply by dbritto-dev** [reply]
https://zustand.docs.pmnd.rs/migrations/migrating-to-v5#persist-middlware-no-longer-stores-item-at-store-creation

**Reply by limuen** [reply]
> [https://zustand.docs.pmnd.rs/migrations/migration-to-v5#persist-middlware-no-longer-stores-item-at-store-creation](https://zustand.docs.pmnd.rs/migrations/migrating-to-v5#persist-middlware-no-longer-stores-item-at-store-creation)

So after upgrading to v5, I don't have useCountStore.setState, but I can still get the initialization status

**Reply by dbritto-dev** [reply]
@limuen after upgrading to v5 you have to use to `setState()` to persist the `initialState`.

```ts
useUserStore.setState(state => state)
```
